src/main/java/org/molgenis/vcf/report/repository/DatabaseManager.java

```diff
@@ -0,0 +1,243 @@
+    }
+
+
+    public Connection getConnection(String databaseLocation) {
+        String databaseUrl = String.format("jdbc:sqlite:%s", databaseLocation);
+        if (this.conn == null) {
+            try {
+                this.conn = DriverManager.getConnection(databaseUrl);
+            } catch (SQLException e) {
+                throw new DatabaseException(e.getMessage(), "get connection");
+            }
+        }
+        return this.conn;
+    }
+
+    public Bytes populateDb(String databaseLocation, FieldMetadatas fieldMetadatas, Items<Sample> samples, File vcfFile,
+                            Path decisionTreePath, Path sampleTreePath, ReportMetadata reportMetadata, Map<?, ?> templateConfig,
+                            @NonNull List<Phenopacket> phenopackets) throws IOException {
+        getConnection(databaseLocation);
+        boolean isGz = vcfFile.getName().toLowerCase().endsWith(".gz");
+        try (
+                InputStream fis = isGz
+                        ? new GZIPInputStream(new FileInputStream(vcfFile))
+                        : new FileInputStream(vcfFile);
+                BufferedReader br = new BufferedReader(new InputStreamReader(fis, StandardCharsets.UTF_8));
+                Utf8LineReader utf8LineReader = new Utf8LineReader(br);
+                LineIteratorImpl vcfIterator = new LineIteratorImpl(utf8LineReader)
+        ) {
+            VCFCodec codec = new VCFCodec();
+            VCFHeader header = (VCFHeader) codec.readActualHeader(vcfIterator);
+                conn.setAutoCommit(false);
+
+                Map<String, FieldMetadata> parentFields = getNestedFields(fieldMetadatas, INFO);
+                Map<String, List<String>> nestedFields = new HashMap<>();
+                for (String field : parentFields.keySet()) {
+                    nestedFields.put(field, getDatabaseNestedColumns(field));
+                }
+                List<String> lines = new ArrayList<>(header.getMetaDataInInputOrder().stream().map(VCFHeaderLine::toString).toList());
+                metadataRepo.insertHeaderLine(conn, lines, getHeaderLine(samples));
+                Map<FieldType, Map<String, Integer>> metadataKeys = metadataRepo.insertMetadata(conn, fieldMetadatas, decisionTreePath, sampleTreePath, phenopackets);
+                sampleRepo.insertSamples(conn, samples);
+                insertVariants(fieldMetadatas, samples, decisionTreePath, sampleTreePath, header, vcfIterator, nestedFields, codec, metadataKeys);
+                phenotypeRepo.insertPhenotypeData(conn, phenopackets, samples.getItems());
+                if(templateConfig != null) {
+                    configRepo.insertConfigData(conn, templateConfig);
+                }
+                decisionTreeRepo.insertDecisionTreeData(conn, decisionTreePath, sampleTreePath);
+                reportMetadataRepo.insertReportMetadata(conn, reportMetadata);
+
+                conn.commit();
+        }catch (SQLException e){
```

Comment:
fix formatting (on project?)

Reply:
I believe this is fixed, not sure why it is not showing as "outdated"